src/components/Table/Table.tsx

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+import { useSearchParams } from 'react-router-dom';
+import cn from 'classnames';
+import { Person } from '../../types';
+import { SearchLink } from '../SearchLink';
+import { PersonRow } from '../PersonRow';
+import { SortOrder } from '../../types/SortOrder';
+import { SortFields } from '../../types/SortFields';
+
+interface Props {
+  displayPeople: Person[]
+}
+
+export const Table: React.FC<Props> = ({
+  displayPeople,
+}) => {
+  const [searchParams] = useSearchParams();
+  const sort = searchParams.get('sort') || '';
+  const sortOrder = searchParams.get('sortOrder') || '';
+
+  const getSortedParams = (
+    newSortType: string,
+  ): Record<string, string | null> => {
+    if (sort !== newSortType) {
+      return {
+        sort: newSortType,
+        sortOrder: SortOrder.Asc,
+      };
+    }
+
+    if (sort === newSortType && sortOrder === SortOrder.Asc) {
+      return { sortOrder: SortOrder.Desc };
+    }
+
+    return {
+      sort: null,
+      sortOrder: null,
+    };
+  };
+
+  const sortingByHeader = (sortField: SortFields) => {
+    return (
+      <SearchLink params={getSortedParams(sortField)}>
+        <span className="icon">
+          <i
+            className={cn('fas', {
+              'fa-sort': sort !== `${sortField}`,
+              'fa-sort-up': sort === `${sortField}` && sortOrder === SortOrder.Asc,
+              'fa-sort-down': sort === `${sortField}` && sortOrder === SortOrder.Desc,
+            })}
+          />
+        </span>
+      </SearchLink>
+    );
+  };
```

Comment:
It looks like a component so rename and create a separate file for it. And I think it could be without link.
```typescript
const SortIcon = ({ sortField }) => {
  const sort = searchParams.get('sort') || '';

  return (
    <span className="icon">
      <i
         className={cn('fas', {
           'fa-sort': sort !== `${sortField}`,
           'fa-sort-up': sort === `${sortField}` && sortOrder === SortOrder.Asc,
           'fa-sort-down': sort === `${sortField}` && sortOrder === SortOrder.Desc,
         })}
       />
    </span>
  );
};
```